src/js/components/AppListComponent.jsx

```diff
@@ -148,105 +148,135 @@ var AppListComponent = React.createClass({
     });
   },
 
-  getGroupedNodes: function (appsSequence) {
-    var currentGroup = this.props.currentGroup;
-
-    return appsSequence
-      .filter(app => app.id.startsWith(currentGroup))
-      .reduce((memo, app) => {
-        let relativePath = app.id.substring(currentGroup.length);
-        let pathParts = relativePath.split("/");
-        let isGroup = pathParts.length > 1;
-
-        if (!isGroup) {
-          memo.push(app);
-        } else {
-          let groupId = currentGroup + pathParts[0];
-          let group = memo.find(item => {
-            return item.id === groupId;
-          });
-
-          if (group == null) {
-            group = initGroupNode(groupId, app);
-            memo.push(group);
-          } else {
-            updateGroupNode(group, app);
-          }
-        }
-        return memo;
-      }, []);
+  hasFilters: function () {
+    return Object.values(this.props.filters).some((filter) => {
+      return (Util.isArray(filter) && filter.length > 0) ||
+        (Util.isString(filter) && filter != null && filter !== "");
+    });
   },
 
-  getAppNodes: function () {
-    var state = this.state;
-    var sortKey = state.sortKey;
+  filterNodes: function (nodesSequence) {
     var props = this.props;
+    var currentGroup = props.currentGroup;
+    var filters = props.filters;
 
-    var appsSequence = lazy(state.apps);
+    if (currentGroup !== "/") {
+      nodesSequence = nodesSequence
+        .filter(app => app.id.startsWith(currentGroup));
+    }
 
-    if (props.filterText != null && props.filterText !== "") {
-      appsSequence = appsSequence
+    if (filters.filterText != null && filters.filterText !== "") {
```

Comment:
It would work, but I want to try to be consistent in this function with our notation. Compared to the other filter checks. OK?